[IMP] views: setting view (#3092)
odoo/documentation: content/developer/reference/backend/views.rst

.. todo:: widgets?

Settings Form View

> ------------------

Reviewer: Shouldn't this section be a sub-section of Form?
```suggestion
~~~~~~~~~~~~~~~~~~
```
The title could be **Settings components** to be consistent with the other sub-sections, for example.

Author: The settings view, is not a component of the form view. It's a jsClass, a customization of the form view, that have a search bar (form view don't have one), a sidebar, and had its own components. It's for these reasons that we put-it as an independent view.

  Parameters:

  .. rst-class:: o-definition-list

  ``type`` (optional)
    By default, a setting is visually separated on two panels (left and right), and is used to
    edit a given field. By defining ``type='header'``, a special kind of setting is rendered
    instead. This setting is used to modify the scope of the other settings. For example, on the
    website application, this setting is used to indicate to which website the other settings
    apply. The header setting is visually represented as a yellow banner on the top of the screen.
  ``string`` (optional)
    The text used as label of the setting. If it's not defined, the first field is used as label.
  ``title`` (optional)
    The text used as tooltip.
  ``help`` (optional)
    The help/description of the setting. This text is displayed just below the setting label (with classname ``text-muted``).
  ``company_dependent`` (optional)
    If this attribute is set to "1" an icon is displayed next to the setting label to explicit
    that this setting is company-specific.
  ``documentation`` (optional)
    If this attribute is set, an icon is added next to the setting label, this icon is a link to the documentation.

>     Note that you can use relative or absolute path. The relative path is relative to ``https://www.odoo.com/documentation/server_version``,

Reviewer: Nitpicking.
```suggestion
    Note that you can use relative or absolute path. The relative path is relative to ``https://www.odoo.com/documentation/<server_version>``,
```

    so it's not necessary to hard-code the server version on the arch anymore.

``Example``

.. code-block:: xml

    <app string="CRM" name="crm">
        <setting type="header" string="Foo">
            <field name="foo" title="Foo?."/>
            <button name="nameAction" type="object" string="Button"/>
        </setting>
        <block title="Title of group Bar">
            <setting help="this is bar" documentation="/applications/technical/web/settings/this_is_a_test.html">
                <field name="bar"/>
            </setting>
            <setting string="This is Big BAR" company_specific="1">
                <field name="bar"/>
            </setting>
        </block>
        <block title="Title of group Foo">
            <setting string="Personalize setting" help="this is full personalize setting">
                <div>This is a different setting</div>
            </setting>
        </block>

>     </app>

Reviewer: Use the `example` directive.
```rst
.. example::

   .. code-block:: XML

       <app string="CRM" name="CRM">
           <setting type="header" string="Foo">
            ...
```

has the effect to activate first ``bar`` then ``foo``.

View customizations
-------------------

As seen in :ref:`Customizing an existing view <reference/js_class>` we can create a custom version
of a generic view. Here we will present some of these views:

> Settings View

Reviewer 2: I think I'd place this section inside the Form View section (thus removing the above paragraph)

~~~~~~~~~~~~~

The settings view is a customization of the form view. It's used to centralize all the settings of Odoo.

> This view differs from a generic form view because it has a search panel and a sidebar.

Reviewer 2: search bar?

To avoid complecity, simplify and standardize the settings archs, a small DSL exclusively for the

> settings was created. This new DSL introduces 3 tags: ``app``, ``block`` and ``setting``.

Reviewer 2: ```suggestion
The settings form view accepts three additional tags: ``app``, ``block`` and ``setting``.
```

.. rst-class:: o-definition-list

``app``
  The ``app`` tag is used to declare the application on the settings view. It creates an entry with
  its logo on the sidebar of the view. It also acts as delimiter when searching.

  Syntax:

  .. code-block:: xml

    <app string="CRM" name="crm">
    ...
    </app>

  Parameters:

  .. rst-class:: o-definition-list

  ``string``
    The "display" name of the application.
  ``name``
    The technical name of the application (the name of the module).
  ``logo`` (optional)
    The relative path to the logo. If not set, the logo is created using the ``name`` parameter : ``/{name}/static/description/icon.png``.

``block``
  The ``block`` tag is used to declare a group of settings. This group can have a title and a description/help.

  Syntax:

  .. code-block:: xml

    <block title="Title of group Bar">
    ...
    </block>

  Parameters:

  .. rst-class:: o-definition-list

  ``title`` (optional)

>     The title of the block of settings (the old h2), you can perform research on its text.

Reviewer 2: in the doc, there should be no notion of "old" (unlike in the commit, where you explain what changes).
